pkg/vault/config.go

```diff
@@ -0,0 +1,54 @@
+package vault
+
+import (
+	"errors"
+	"os"
+)
+
+// Config TODO
+type Config struct {
+	Address string
+	Type    VaultType
+}
+
+// NewConfig returns a new Config struct
+func NewConfig() (*Config, error) {
+	config := &Config{
+		Address: getEnv("VAULT_ADDR", ""),
+	}
+
+	switch getEnv("VAULT_TYPE", "") {
+	case "vault":
+		auth := getEnv("AUTH_TYPE", "")
+		switch auth {
+		case "approle":
+			config.Type = &AppRole{
+				RoleID:   getEnv("ROLE_ID", ""),
+				SecretID: getEnv("SECRET_ID", ""),
+			}
+		case "github":
+			config.Type = &Github{
+				AccessToken: getEnv("GITHUB_TOKEN", ""),
+			}
+		default:
+			return nil, errors.New("Must provide a supported Athentication Type")
```

Comment:
*Authentication

I once got a warning saying that error messages shouldn't start with a capital letter. No idea if that's accepted practice or not